drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,831 @@
+						ESPI_SLAVE_TO_MASTER },
+	/* MSVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_WARN]      = { MSVW03, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SUS_PWRDN_ACK] = { MSVW03, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SLP_A]         = { MSVW03, ESPI_VWIRE_SRC_ID3,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW04 */
+	[ESPI_VWIRE_SIGNAL_SLP_LAN]       = { MSVW04, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	[ESPI_VWIRE_SIGNAL_SLP_WLAN]      = { MSVW04, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW07 */
+	[ESPI_VWIRE_SIGNAL_HOST_C10]      = { MSVW07, ESPI_VWIRE_SRC_ID0,
+						ESPI_MASTER_TO_SLAVE },
+	/* MSVW08 */
+	[ESPI_VWIRE_SIGNAL_DNX_WARN]      = { MSVW08, ESPI_VWIRE_SRC_ID1,
+						ESPI_MASTER_TO_SLAVE },
+};
+
+static int espi_xec_configure(struct device *dev, struct espi_cfg *cfg)
+{
+	u8_t iomode = 0;
+	u8_t cap1 = ESPI_CAP_REGS->GLB_CAP1;
+	u8_t cur_iomode = (cap1 & MCHP_ESPI_GBL_CAP1_IO_MODE_MASK) >>
+		MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+
+	/* Set frequency */
+	cap1 &= ~MCHP_ESPI_GBL_CAP1_MAX_FREQ_MASK;
+	switch (cfg->max_freq) {
+	case 20:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_20M;
+		break;
+	case 25:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_25M;
+		break;
+	case 33:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_33M;
+		break;
+	case 50:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_50M;
+		break;
+	case 66:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_66M;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Set IO mode */
+	iomode = (cfg->io_caps >> 1);
+	if (iomode > 3) {
+		return -EINVAL;
+	}
+
+	if (iomode != cur_iomode) {
+		cap1 &= ~MCHP_ESPI_GBL_CAP1_IO_MODE_MASK0 <<
+			MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+		cap1 |= (iomode << MCHP_ESPI_GBL_CAP1_IO_MODE_POS);
+	}
+
+	ESPI_CAP_REGS->GLB_CAP1 = cap1;
+
+	/* Activate the eSPI block *.
+	 * Need to guarantee that this register is configured before RSMRST#
+	 * de-assertion and after pinmux
+	 */
+	ESPI_EIO_BAR_REGS->IO_ACTV = 1;
+	LOG_DBG("eSPI block activated successfully\n");
+	return 0;
+}
+
+static bool espi_xec_channel_ready(struct device *dev,
+					enum espi_channel ch)
+{
+	bool sts;
+
+	switch (ch) {
+	case ESPI_CHANNEL_PERIPHERAL:
+		sts = ESPI_CAP_REGS->PC_RDY & MCHP_ESPI_PC_READY;
+		break;
+	case ESPI_CHANNEL_VWIRE:
+		sts = ESPI_CAP_REGS->VW_RDY & MCHP_ESPI_VW_READY;
+		break;
+	case ESPI_CHANNEL_OOB:
+		sts = ESPI_CAP_REGS->OOB_RDY & MCHP_ESPI_OOB_READY;
+		break;
+	case ESPI_CHANNEL_FLASH:
+		sts = ESPI_CAP_REGS->FC_RDY & MCHP_ESPI_PC_READY;
+		break;
+	default:
+		sts = false;
+		break;
+	}
+
+	return sts;
+}
+
+static int espi_xec_send_vwire(struct device *dev,
+				enum espi_vwire_signal signal,
+				u8_t level)
+{
+	struct xec_signal signal_info = vw_tbl[signal];
+	u8_t xec_id = signal_info.xec_reg_idx;
+	u8_t ESPI_VWIRE_SRC_ID = signal_info.bit;
+
+	if ((ESPI_VWIRE_SRC_ID >= ESPI_VWIRE_SRC_ID_MAX) ||
+		(xec_id >= ESPI_MSVW_IDX_MAX)) {
+		return -EINVAL;
+	}
+	if (signal_info.dir == ESPI_MASTER_TO_SLAVE) {
```

Comment:
always add an empty line after '}' unless it's another '}' etc... Apply that everywhere.

Reply:
Addressed